### ISSUE TITLE
How to force extension to respect local .vscode folder beneath parent?

### ISSUE THREAD

**Issue by artisticcheese** [mkloubert/vs-deploy#8]
Hello,

I open VSCode folder say c:\projects\ which has subfolders \project1, \project2 etc.
Extension seems to be looking for settings.json under c:\projects while .vscode folder under \project1 etc being ignored. How do I define destinations for specific subfolders?

**Comment by mkloubert** [comment]
The extension uses the API of VS Code to read the settings. The problem is, that VS Code only searches for the direct occurrence of `.vscode` subfolder in your workspace's root directory. To search deeper inside the hierachie, it takes too much time for Visual Studio.

But you can do the following:

Simply define two packages:

```json
{
    "deploy": {
        "packages": [
            {
                "name": "Project 1",
                "description": "The files of 'project1' subfolder",
                "files": [
                    "project1/**"
                ]
            },

            {
                "name": "Project 2",
                "description": "The files of 'project2' subfolder",
                "files": [
                    "project2/**"
                ]
            }
        ]
    }
}
```

But this brings me to the idea, to implement a new feature: Import settings from other locations.

Until the feature is implemented, you should use the "package solution".

**Comment by artisticcheese** [comment]
How do I define different targets for different packages?

**Comment by mkloubert** [comment]
"Targets" and "packages" are two different things.

The idea is:

Packages describe a subset of your workspace's files, which should be deployed.

The targets describe where one file (or a package) should be deployed to.

In the editor (when you press `CTRL + ALT + W` or select `Deploy: Workspace` command), you first select the package and after that you select the target.

**Comment by artisticcheese** [comment]
Right now I just do "deploy" and it automatically deploys, it does not prompt me for target, might be because I have only 1 target and 1 package? I expected to have dictionary of package/target to able to deploy file to destination with single command.

**Comment by mkloubert** [comment]
Thats right.

If you only have one package, there will be no prompt ... It automatically selects the only package.

The same thing happens with targets.

If you have only one target AND only one package, it directly starts deploying.

**Comment by artisticcheese** [comment]
So what shall I do if I have workspace with different subfolders which shall be deployed to different targets? I really hoped that target will actually be part of "package", so you can have one click deployment for any source files.

**Comment by mkloubert** [comment]
This could become a new feature ... that you can define one or more targets inside a package if you want ... so you become able to deploy a package to target(s) directly.

I think this is a good chance to implement this in the next version of the extension.

**Comment by artisticcheese** [comment]
I think you can just make a new field "targets" in "package" which will reference one of the "targets" objects. So no need to create entire "target" object for each individual package.

**Comment by mkloubert** [comment]
This is what I want to implement.

The new field will be only a string (or a list of strings), which are references to defined target(s).

**Comment by mkloubert** [comment]
**Implemented since Version 3.18.0!**

Now you can define explicit targets in your package, if you want:

```json
{
    "deploy": {
        "packages": [
            {
                "name": "Project 1",
                "description": "The files of 'project1' subfolder",
                "files": [
                    "project1/**"
                ],

                "targets": [
                    "My ZIP file", "My SFTP"
                ]
            }
        ],

        "targets": [
            {
                "name": "My ZIP file",
                "type": "zip",
                "target": "E:/test/deploy/zip"
            },
            
            {
                "name": "My SFTP",
                "type": "sftp",
                "host": "sftp.example.com",
                "user": "mkloubert", "password": "P@assword123!",
                "dir": "/"
            }
        ]
    }
}
```
